rdm/audit_for_gaps.py

```diff
@@ -175,7 +178,69 @@ def _sorted_checklist_items(unsorted_checklist):
         key = checklist_item.get('reference', '')
```

Comment:
@orwonthe when would the reference be a blank string? Would that be true for comments?

Reply:
Currently the method is only used to sort checklists where the includes and comments are removed. So the reference is always there. The code here is cautiously allowing the reference to be absent.

Reply:
This is an interesting coding question.

It seems to me like we would want it to throw an error if the reference isn't there and _we expect it to be there_.

Also, as written the code implies that it is possible for the reference to be missing when it actually isn't, which IMO makes it harder to read (a comment stating this is defensive could address this, but per my earlier comment that seems unnecessary).

Note that it isn't a big deal if this code errors out.

What do you think of this reasoning?

Philosophically I am saying that it I don't think this is a good place to code "defensively".

Reply:
Agreed

Reply:
Fixed.